buildcontext/gitlookup.go

```diff
@@ -246,6 +248,10 @@ func (gl *GitLookup) lookupNetRCCredential(host string) (login, password string,
 	if err != nil {
 		return "", "", errors.Wrap(err, "failed to parse .netrc")
 	}
+	machine := n.Machine(host)
+	if machine == nil {
+		return "", "", errors.Wrapf(errNoRCHostEntry, "failed to lookup netrc entry for %s", host)
```

Comment:
Would this case cause it to fall back to other auth mechanisms?

Reply:
it falls back to unauthenticated https, as the error messages are actually ignored: https://github.com/earthly/earthly/blob/fix-panic-when-netrc-entry-missing/buildcontext/gitlookup.go#L304

Reply:
and this is only when the auth is set to "auto" (the default), first it tries connecting via ssh, then netrc, then un-authed.

Reply:
great